packages/core/admin/admin/src/content-manager/components/BlocksEditor/Toolbar/index.js

```diff
@@ -464,21 +464,58 @@ const isListActive = (editor, matchNode) => {
 };
 
 const toggleList = (editor, isActive, format) => {
-  // Delete the parent list so that we're left with only the list items directly
-  Transforms.unwrapNodes(editor, {
-    match: (node) => isListNode(node) && ['ordered', 'unordered'].includes(node.format),
-    split: true,
-  });
+  /**
+   * If we have selected a portion of content in the editor, we want to convert it to a list or if it is already a list, convert it back to a paragraph
+   */
+  if (editor.selection) {
+    Transforms.unwrapNodes(editor, {
+      match: (node) => isListNode(node) && ['ordered', 'unordered'].includes(node.format),
+      split: true,
+    });
 
-  // Change the type of the current selection
-  Transforms.setNodes(editor, {
-    type: isActive ? 'paragraph' : 'list-item',
-  });
+    Transforms.setNodes(editor, {
+      type: isActive ? 'paragraph' : 'list-item',
+    });
+
+    if (!isActive) {
+      const block = { type: 'list', format, children: [] };
+      Transforms.wrapNodes(editor, block);
+    }
+    /**
+     * Otherwise if there is not a selected portion in the editor, we want to convert the last inserted node to a list or if it is already a list, convert it back to a paragraph
+     */
+  } else {
```

Comment:
can you move the comment inside of the else block?

also, please avoid having lines of comments that are longer than the lines of code, as they're harder to read in an IDE. For this one you could split it like this for example:

```js
// There is no selection, convert the last inserted node to a list
// If it is already a list, convert it back to a paragraph
```